web-common/src/components/search/Search.svelte

```diff
@@ -36,6 +37,27 @@
     // Keep ref optional here. If component is unmounted before this animation frame runs, ref will be null and throw a TypeError
     if (autofocus) window.requestAnimationFrame(() => ref?.focus());
   });
+
+  const BaseHeight = 28;
+  const HeightPerLine = 16;
+  const MaxLines = 5;
+
+  // For tailwind compiler: h-[28px] h-[44px] h-[60px] h-[76px] h-[92px]
+  let height = "h-[28px]";
+  function updateHeight(value: string) {
+    const lines = value.split("\n").length;
+    const correctedLines = Math.max(
+      // Show minimum of 1 line
+      1,
+      // We expand the input only till MaxLines
+      Math.min(lines, MaxLines),
+    );
+
+    height = `h-[${BaseHeight + (correctedLines - 1) * HeightPerLine}px]`;
+    console.log(lines, correctedLines, height);
+  }
```

Comment:
Can we use CSS rather than JS for this?

Reply:
You mean something like `h-7` `h-10` etc? Not all height values we use here have it, otherwise it would have been perfect.

Reply:
I mean using a class something like this:
```css
.textarea-auto-height {
  height: auto;
  min-height: 28px;
  max-height: 92px; /* Equivalent to 5 lines: 28px + (4 * 16px) */
  line-height: 16px;
  overflow-y: auto;
  resize: none;
  box-sizing: border-box;
  padding: 6px 8px;
}
```

Reply:
Updated to use both js and min/max height.